test/suite/wttest.py

```diff
@@ -457,17 +459,18 @@ def tearDown(self):
             for f in files:
                 os.chmod(os.path.join(root, f), 0o666)
         self.pr('passed=' + str(passed))
+        self.pr('skipped=' + str(self.skipped))
 
         # Clean up unless there's a failure
-        if passed and not WiredTigerTestCase._preserveFiles:
+        if passed and not WiredTigerTestCase._preserveFiles or self.skipped:
```

Comment:
The code is correct, but I would add parentheses around the `and` part of the expression.   I never fully remember the order, and every time I see the code, I have to think about it.  Adding parentheses reduces the cognitive load.

Reply:
Readability counts, reworked!